File: view/view-farma/src/app/interface/output-product.ts
import { OutPut } from "./out-put"; // Importe a interface de OutPut

export interface OutputProduct {
  id?: number; // Em TypeScript, 'number' corresponde a Long e outras formas de números inteiros
  output?: OutPut; // Relacionamento ManyToOne com OutPut
  codLot?: string;
  description?: string;
  unit?: string;
  amount?: number; // 'int' em Java corresponde a 'number' em TypeScript
}

File: src/main/java/com/horus/formataArquivoHorus/model/entity/OutputProduct.java
package com.horus.formataArquivoHorus.model.entity;

import jakarta.persistence.*;
import lombok.Data;

@Entity
@Table(name = "output_product_tb")
@Data
public class OutputProduct {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "id")
    private Long id;

    @ManyToOne
    @JoinColumn(name = "output_id", referencedColumnName = "id")
    private OutPut output;

    @Column(name = "codLot")
    private String codLot;

    @Column(name = "description")
    private String description;

    @Column(name = "unit")
    private String unit;

    @Column(name = "amount")
    private int amount;
}

